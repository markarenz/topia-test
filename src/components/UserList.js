import React from "react";
import Container from "@material-ui/core/Container";
import Typography from "@material-ui/core/Typography";
import Box from "@material-ui/core/Box";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import TableCell from "@material-ui/core/TableCell";
import BroadcastIndicator from "./BroadcastIndicator";
import { useUsersContext } from "../context/UsersContext";
import { DISPLAY_MODES } from "../utils/constants";

const UserList = () => {
  const { displayMode, usersInView } = useUsersContext()

  if (displayMode !== DISPLAY_MODES.USER_LIST) {
    return null
  }

  return (
    <Container maxWidth="md">
      <Box pt={12} mb={5}>
        <Typography variant="h5" component="h1" gutterBottom>
          The following Users are currently visible based on position and screen size.
        </Typography>
      </Box>
      {
        usersInView.length === 0
          ? (
            <Box my={4}>
              <Typography component="p" gutterBottom>
                There are currently no users within view.
              </Typography>
            </Box>
          ) : (
            <Container maxWidth="sm">
              <Table>
                <TableHead color="primary">
                  <TableRow>
                    <TableCell>
                      <Typography variant='h6'>
                        Username
                      </Typography>
                    </TableCell>
                    <TableCell>
                      <Typography variant='h6'>
                        Distance
                      </Typography>
                    </TableCell>
                    <TableCell align="right">
                      <Typography variant='h6'>
                        Is Broadcaster
                      </Typography>
                    </TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {
                    usersInView.map((user) => (
                      <TableRow>
                        <TableCell>
                          {/* Add ellipsis just in case we have long usernames in the future */}
                          <Typography
                            variant='body1'
                            style={{
                              maxWidth: '200px',
                              overflow: 'hidden',
                              textOverflow: 'ellipsis',
                              whiteSpace: 'nowrap',
                            }}
                          >
                            {user.username}
                          </Typography>
                        </TableCell>
                        <TableCell>
                          <Typography variant='body1'>
                            {Math.floor(user.distanceToPlayer * 100) / 100}
                          </Typography>
                        </TableCell>
                        <TableCell>
                            {
                              user.is_broadcaster && (
                                <Box
                                  sx={{ display: 'flex', justifyContent: 'end'}}
                                >
                                  <BroadcastIndicator />
                                </Box>
                              )
                            }
                        </TableCell>
                      </TableRow>
                    ))
                  }
                </TableBody>
              </Table>
            </Container>
          )
      }
    </Container>
  );
};

export default UserList;