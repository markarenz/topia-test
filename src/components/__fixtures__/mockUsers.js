export const mockUsers = [
  {
    id: 2,
    is_broadcaster: false,
    username: "user2",
    x: 0,
    y: 0,
    distanceToPlayer: 200
  },
  {
    id: 1,
    is_broadcaster: false,
    username: "user1",
    x: 3500,
    y: 150,
     distanceToPlayer: 900
 },
]