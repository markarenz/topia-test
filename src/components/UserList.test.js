import React from "react";
import { render } from "@testing-library/react";
import UserList from "./UserList";
import { mockUsers } from "./__fixtures__/mockUsers";

jest.mock('../context/UsersContext', () => ({
  useUsersContext: () => ({
    displayMode: 'userList',
    usersInView: mockUsers
  })
}));

describe("<UserList />", () => {

  it("renders correctly", () => {
    jest.mock('../context/UsersContext', () => ({
      useUsersContext: () => ({
        displayMode: 'userList',
        usersInView: mockUsers
      })
    }));
    const {container} = render(
      <UserList />
    );
    expect(container).toMatchSnapshot();
  });
});
