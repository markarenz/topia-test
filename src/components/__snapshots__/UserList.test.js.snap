// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> renders correctly 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <h1
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
      >
        The following Users are currently visible based on position and screen size.
      </h1>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
          color="primary"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                Username
              </h6>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                Distance
              </h6>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
              scope="col"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6"
              >
                Is Broadcaster
              </h6>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
                style="max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
              >
                user2
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                200
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            />
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
                style="max-width: 200px; overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
              >
                user1
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                900
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
